src/components/Primary-Button/Button.styles.ts

```diff
@@ -0,0 +1,68 @@
+import { Button as MuiButton, CircularProgress } from '@mui/material';
+import { styled } from '@mui/material/styles';
+
+import { SIZE_CONFIGS, VARIANT_COLORS, SHAPE_CONFIGS } from './Button.constants';
+
+import type { ButtonProps } from './Button.types';
+
+export const LoadingSpinner = styled(CircularProgress)(() => ({
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  color: 'inherit',
+}));
+
+export const ButtonContent = styled('span')<{ $loading: boolean }>(({ $loading }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  gap: '8px',
+  opacity: $loading ? 0 : 1,
+  transition: 'opacity 0.3s ease',
+}));
+
+export const StyledButton = styled(MuiButton)<ButtonProps>(({
+  theme,
+  variant,
+  size,
+  shape,
+  loading,
+  fullWidth,
+  disabled,
+}) => {
+  const colors = VARIANT_COLORS[variant || 'primary'];
+  const sizeConfig = SIZE_CONFIGS[size || 'medium'];
+  const shapeConfig = SHAPE_CONFIGS[shape || 'rounded'];
+
+  return {
+    height: sizeConfig.height,
+    minWidth: fullWidth ? '100%' : sizeConfig.minWidth,
+    padding: theme.spacing(sizeConfig.paddingScale[0], sizeConfig.paddingScale[1]),
+    fontSize: `${sizeConfig.fontSizeRem}rem`,
+    fontFamily: 'inherit',
+    fontWeight: 500,
+    borderRadius: shapeConfig,
+    background: variant === 'primary' ? colors.main : '#fff',
```

Comment:
The styles include hard-coded colors (e.g. '#fff') and reference VARIANT_COLORS which themselves hard-code hex strings/gradients. Per project guidelines, colors and backgrounds should come from the theme (theme.palette or theme.custom tokens). Using '#fff' and similar literals here prevents consistent theming and may break dark mode/contrast requirements.

Suggestion:
- Replace hard-coded values with theme tokens. Example:
  background: variant === 'primary' ? theme.gradients.primary : theme.palette.background.paper;
  color: variant === 'primary' ? theme.palette.primary.contrastText : theme.palette.primary.main;
- If VARIANT_COLORS must remain, at least map them through the theme in the styles layer, not as raw strings in constants.